### ISSUE TITLE
Issue 55 - Update test all graphs code to read from file with variable buffer size

### ISSUE THREAD

**Issue by wbkboyer** [graph-algorithms/edge-addition-planarity-suite#55]
Amend the test all graphs code to change from the current `ReadTextFileIntoString()` technique to instead read from file pointer.

Add a hidden command-line parameter to indicate the buffer size, to be used with [`setvbuf()`](https://en.cppreference.com/w/c/io/setvbuf). If the command line parameter is not supplied, less than `BUFSIZ`, or greater than 2 billion, then don't call `setvbuf()` (use default buffer size).

Later, we can test to see if there is a benefit to using [`fstat()`](https://www.gnu.org/software/libc/manual/html_node/Reading-Attributes.html) to determine the [`st_blksize`](https://www.gnu.org/software/libc/manual/html_node/Attribute-Meanings.html) to determine the optimal buffer size for reading a given file.

These are the timing tests we'd like to try (with the release version, for best speed): 

1) With default `fopen()` behavior, on SSD and on HDD, what is the total read-and-process time of planarizing all graphs in a single large file such as all 11 vertex twenty-some-odd-edge graphs?
2) With `fopen()` plus `setvbuf()` to a very large size given by a command-line parameter, on SSD and on HDD, what is the total read-and-process time of planarizing all graphs in a single large file such as all 11 vertex twenty-some-odd-edge graphs?
3) With `fopen()` plus `setvbuf()` to a size recommended by `fstat()` block size, on SSD and on HDD, what is the total read-and-process time of planarizing all graphs in a single large file such as all 11 vertex twenty-some-odd-edge graphs?

We hope to get the best configuration so that we can run 6 cores for each separate algorithm to test all 11-vertex graph files (separate story).

**Comment by wbkboyer** [comment]
TL;DR Created #60 to read from `FILE` pointer *without* manipulating the internal buffer size, since it doesn't improve performance reading from file pointer.

On my Windows 10 PC, I obtained the following results for running the `planarity` release build on the `test.n11.m27.g6` input file (~1.4GB, generated using `geng 11 27:27`):

|              | ReadTextFileIntoString() | `fopen()` with default buffer size | `fopen()` with 1.5GB buffer size | `fopen()` with `BUFFSIZ` (512 on Windows) |
|--------------|--------------------------|------------------------------------|---------------------------------|-----------------------|
| HDD          | 376.140s (~110 MB/s) | 371.109s (3.8 MB/s) | 380.859s (110 MB/s) | 370.812s (3.5 MB/s) |
| external SSD | 394.797s (~45 MB/s) | 371.063s (3.5 MB/s) | 411.125s (45 MB/s) | 369.422s (3.5 MB/s)|

*N.B.* I didn't test reading from my Windows desktop's *internal* SSD, as it doesn't have sufficient space to perform testing of the larger `.g6` files; the external SSD, despite being connected to the USB 3 port (both on front and back of case), had pretty disappointing read results, but reading doesn't really seem to be the bottleneck.

For curiosity's sake, I decided to test on MacOS: `fopen()` with default buffer size took 200.357 seconds, whereas `fopen()` with calling `setvbuf` using the value from `stat.st_blksize` took 198.732 seconds.
